components/infobox/extensions/wikis/stormgate/infobox_extension_attack.lua

```diff
@@ -0,0 +1,98 @@
+---
+-- @Liquipedia
+-- wiki=stormgate
+-- page=Module:Infobox/Extension/Attack
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Json = require('Module:Json')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+local Title = Widgets.Title
+
+local Attack = {}
+
+---@class StormgateAttackData
+---@field name string
+---@field targets string[]?
+---@field damage number?
+---@field damagePercentage string
+---@field effect string
+---@field speed number?
+---@field dps number?
+---@field bonus string
+---@field bonusDamage number?
+---@field bonusDps number?
+---@field range number?
+
+---@param argsJson string
+---@param attackIndex integer
+---@param faction string
+---@return Widget[]?
+function Attack.run(argsJson, attackIndex, faction)
+	local args = Json.parseIfTable(argsJson)
+	if not args then return end
+	assert(args.name, 'Please specify a name for attack ' .. attackIndex)
+	assert(faction, 'Faction needs to be specified')
```

Comment:
Where would these appear, appended to the usual text saying "mention this in #bugs"?
Then IMO make this one a bit more specific, e.g. include the name and that it's an attack

Reply:
the idea is to inherit faction from the infobox, hence the error just stating that it needs to be set